components/aws/sagemaker/tests/integration_tests/utils/minio_utils.py

```diff
@@ -5,12 +5,15 @@
 
 
 def get_artifact_in_minio(workflow_json, step_name, artifact_name, output_dir):
+    """ Minio is the S3 style object storage server for K8s. This method parses a pipeline run's workflow json to
+    fetch the output artifact location in Minio server for given step in the pipeline and downloads it"""
+
     s3_data = {}
     minio_access_key = "minio"
     minio_secret_key = "minio123"
     minio_port = utils.get_minio_service_port()
     for node in workflow_json["status"]["nodes"].values():
-        if step_name in node["name"]:
+        if step_name in node["name"] and node["type"] != "DAG":
```

Comment:
What is the meaning of second condition `node["type"] != "DAG"` ?

Reply:
There are two types of nodes in the workflow_json: DAG and pod. DAG corresonds to the whole pipeline and pod corresponds to a step in the DAG. Check `node["type"] != "DAG"` deals with case where name of component is part of the pipeline name.

added it to comments of the method